### PR TITLE
Avoid local network communication / invalid url requests

### PR DESCRIPTION
**Pull request by annando**
This PR covers two related issues:
- It should handle all conditions in which we performed HTTP requests with invalid addresses
- local network communication (when a Friendica server speaks with itself via HTTP requests) is massively reduced

### CODE REVIEW

File: src/Module/Photo.php [hunk 1]
```diff
@@ -349,6 +349,13 @@ private static function getPhotoById(int $id, string $type, int $customsize)
 				} elseif (!empty($contact['avatar'])) {
 					$url = $contact['avatar'];
 				}
+
+				// If it is a local link, we save resources by just redirecting to it.
+				if (Network::isLocalLink($url)) {
+					System::externalRedirect($url);
+					System::exit();
```
Comment (MrPetovan) [inline src/Module/Photo.php]: Superfluous, `System::externalRedirect` already exits.
```suggestion
```

File: src/Module/Photo.php [hunk 2]
```diff
@@ -393,6 +400,10 @@ private static function getPhotoById(int $id, string $type, int $customsize)
 					} else {
 						$url = Contact::getDefaultAvatar($contact ?: [], Proxy::SIZE_SMALL);
 					}
+					if (Network::isLocalLink($url)) {
+						System::externalRedirect($url);
+						System::exit();
```
Comment (MrPetovan) [inline src/Module/Photo.php]:
```suggestion
```

File: src/Module/Photo.php [hunk 3]
```diff
@@ -401,21 +412,30 @@ private static function getPhotoById(int $id, string $type, int $customsize)
...
 				If (($contact['uid'] != 0) && empty($contact['header'])) {
 					$contact = Contact::getByURL($contact['url'], false, $fields);
 				}
 				if (!empty($contact['header'])) {
 					$url = $contact['header'];
 				} else {
 					$url = Contact::getDefaultHeader($contact);
+					if (Network::isLocalLink($url)) {
+						System::externalRedirect($url);
+						System::exit();
```
Comment (MrPetovan) [inline src/Module/Photo.php]:
```suggestion
```

File: src/Module/Photo.php [hunk 4]
```diff
@@ -445,6 +465,11 @@ private static function getPhotoById(int $id, string $type, int $customsize)
 					$default = Contact::getDefaultAvatar($contact, Proxy::SIZE_THUMB);
 			}
 
+			if (Network::isLocalLink($default)) {
+				System::externalRedirect($default);
+				System::exit();
```
Comment (MrPetovan) [inline src/Module/Photo.php]:
```suggestion
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Module/Photo.php (hunk 1 above)
Comment (annando) [reply src/Module/Photo.php]: Oh yeah ...

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**